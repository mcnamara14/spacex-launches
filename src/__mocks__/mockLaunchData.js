export const mockLaunchData = [{
    "flight_number": 1,
    "mission_name": "FalconSat",
    "upcoming": false,
    "launch_year": "2006",
    "launch_date_unix": 1143239400,
    "launch_date_utc": "2006-03-24T22:30:00.000Z",
    "rocket": {
        "rocket_name": "Falcon 1",
        "first_stage": {
            "cores": [
                {
                    "reused": false,
                    "land_success": false,
                }
            ]
        },
    },
    "reuse": {
        "core": false,
        "side_core1": false,
        "side_core2": false,
        "fairings": false,
        "capsule": false
    },
    "links": {
        "mission_patch": "https://images2.imgbox.com/40/e3/GypSkayF_o.png",
        "mission_patch_small": "https://images2.imgbox.com/3c/0e/T8iJcSN3_o.png",
        "article_link": "https://www.space.com/2196-spacex-inaugural-falcon-1-rocket-lost-launch.html",
        "wikipedia": "https://en.wikipedia.org/wiki/DemoSat",
        "video_link": "https://www.youtube.com/watch?v=0a_00nJ_Y88"
    },
  },
  {
  "flight_number": 2,
  "mission_name": "FalconSun",
  "upcoming": false,
  "launch_year": "2008",
  "launch_date_unix": 1143239500,
  "launch_date_utc": "2008-03-24T22:30:00.000Z",
  "rocket": {
    "rocket_name": "Falcon 2",
    "first_stage": {
        "cores": [
            {
                "reused": true,
                "land_success": false,
            }
        ]
    },
  },
  "reuse": {
    "core": false,
    "side_core1": false,
    "side_core2": false,
    "fairings": false,
    "capsule": false
  },
  "links": {
    "mission_patch": "https://images2.imgbox.com/40/e3/GypSkayF_o.png",
    "mission_patch_small": "https://images2.imgbox.com/3c/0e/T8iJcSN3_o.png",
    "article_link": "https://www.space.com/2196-spacex-inaugural-falcon-1-rocket-lost-launch.html",
    "wikipedia": "https://en.wikipedia.org/wiki/DemoSat",
    "video_link": "https://www.youtube.com/watch?v=0a_00nJ_Y88"
  }
}];