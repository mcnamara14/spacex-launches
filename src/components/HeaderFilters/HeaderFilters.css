.HeaderFilters {
  height: 37px;
  padding: 20px;
  display: flex;
  border-radius: 10px 10px 0 0;
  background: rgba(255, 255, 255, 0.15);
  border-left: 1px solid #fff;
  border-top: 1px solid #fff;
  border-right: 1px solid #fff;
  align-items: center;
  justify-content: space-between;
}

.refreshIcon img {
  height: 20px;
  padding: 5px;
  border: 2px solid #fff;
  border-radius: 50%;
}

.checkboxes {
  width: 37%;
  justify-content: space-between;
  display: flex;
}

label {
  display: flex;
  align-items: center;
  color: #fff;
  flex-direction: row-reverse;
  font-size: 14px;
  font-weight: 100;
  position: relative;
}

input[type='checkbox'] {
  margin-right: 10px;
}

.container input {
  position: absolute;
  opacity: 0;
  cursor: pointer;
}

.checkmark {
  position: absolute;
  top: -2px;
  left: -30px;
  height: 18px;
  width: 18px;
  background-color: transparent;
  border: 1px solid;
  border-radius: 3px;
}

.container:hover input ~ .checkmark {
  cursor: pointer;
}

.checkmark:after {
  content: "";
  position: absolute;
  display: none;
}

.container input:checked ~ .checkmark:after {
  display: block;
}

.container .checkmark:after {
  left: 6px;
  top: 1px;
  width: 5px;
  height: 10px;
  border: solid #2196F3;
  border-width: 0 3px 3px 0;
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}