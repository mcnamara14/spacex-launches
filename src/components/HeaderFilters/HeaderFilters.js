import React, { Component } from 'react';
import { connect } from 'react-redux';
import './HeaderFilters.css';
import refreshIcon from './images/refresh.svg';
import { storeLandSuccess } from '../../actions/storeLandSuccess';
import { storeReused } from '../../actions/storeReused';
import { storeReddit } from '../../actions/storeReddit';

class HeaderFilters extends Component {
  toggleLandSuccessFilter = (event) => {
    this.props.landSuccess ? this.props.storeLandSuccess(false) : this.props.storeLandSuccess(true);
  }

  toggleReusedFilter = (event) => {
    this.props.reused ? this.props.storeReused(false) : this.props.storeReused(true);
  }

  toggleRedditFilter = (event) => {
    this.props.reddit ? this.props.storeReddit(false) : this.props.storeReddit(true);
  }

  render() {
    return (
      <section className="HeaderFilters" >
            <div className="refreshIcon"><img src={refreshIcon} /></div>
            <div className="checkboxes">
              <label class="container landSuccess">LAND SUCCESS
                <input type="checkbox" />
                <span class="checkmark" onClick={() => this.toggleLandSuccessFilter()} ></span>
              </label>
              <label class="container reused">REUSED
                <input type="checkbox" />
                <span class="checkmark" onClick={() => this.toggleReusedFilter()}></span>
              </label>
              <label class="container withReddit">WITH REDDIT
                <input type="checkbox" />
                <span class="checkmark" onClick={() => this.toggleRedditFilter()}></span>
              </label>
            </div>
      </section>
    );
  }
}

export const mapDispatchToProps = (dispatch) => ({
  storeLandSuccess: (boolean) => dispatch(storeLandSuccess(boolean)),
  storeReused: (boolean) => dispatch(storeReused(boolean)),
  storeReddit: (boolean) => dispatch(storeReddit(boolean))
});

export const mapStateToProps = (state) => ({
  filters: state.filters,
  landSuccess: state.landSuccess,
  reused: state.reused,
  reddit: state.reddit
});

export default connect(mapStateToProps, mapDispatchToProps)(HeaderFilters);
