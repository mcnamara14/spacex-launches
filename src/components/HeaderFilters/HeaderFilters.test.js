import React from 'react';
import { HeaderFilters } from './HeaderFilters';
import { shallow } from 'enzyme';
import { cleanMockLaunchData } from '../../__mocks__/mockLaunchData';

describe('HeaderFilters', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<HeaderFilters 
      clearFilteredLaunchIds={jest.fn()} 
      storeFilteredLaunchIds={jest.fn()}
      launches={cleanMockLaunchData}/> );
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('toggleFilter', () => {
    beforeEach(() => {
      wrapper.instance().filterLaunches = jest.fn();
    })

    it('should add filter to state when not included already', () => {
      wrapper.setState({filters: ['reused']})

      expect(wrapper.state().filters).toEqual(['reused'])

      wrapper.instance().toggleFilter('reddit')

      expect(wrapper.state().filters).toEqual(['reused', 'reddit'])
    })

    it('should remove filter from state when it is included already', () => {
      wrapper.setState({filters: ['reused', 'reddit']})

      expect(wrapper.state().filters).toEqual(['reused', 'reddit'])

      wrapper.instance().toggleFilter('reddit')

      expect(wrapper.state().filters).toEqual(['reused'])
    })
  })

  describe('filterLaunches', () => {
    it('should store all launches if there are no filters', () => {
      wrapper.instance().filterLaunches()

      expect(wrapper.instance().props.storeFilteredLaunchIds).toHaveBeenCalledTimes(2)
    }) 

    it('should store only launches matching filters', () => {
      wrapper.setState({filters: ['reused']})
      wrapper.instance().filterLaunches()

      expect(wrapper.instance().props.storeFilteredLaunchIds).toHaveBeenCalledTimes(1)
    }) 
  })
});