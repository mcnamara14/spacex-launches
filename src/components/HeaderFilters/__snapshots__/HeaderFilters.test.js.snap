// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderFilters should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HeaderFilters
    clearFilteredLaunchIds={[Function]}
    launches={
        Array [
            Object {
              "article": "https://www.space.com/2196-spacex-inaugural-falcon-1-rocket-lost-launch.html",
              "badge": "https://images2.imgbox.com/3c/0e/T8iJcSN3_o.png",
              "date": "03/24/2006",
              "details": "Engine failure at 33 seconds and loss of vehicle",
              "id": 1,
              "landSuccess": false,
              "name": "Falcon 1",
              "reddit": false,
              "reused": false,
              "type": "Merlin A",
            },
            Object {
              "article": "https://www.space.com/2196-spacex-inaugural-falcon-1-rocket-lost-launch.html",
              "badge": "https://images2.imgbox.com/3c/0e/T8iJcSN3_o.png",
              "date": "03/24/2008",
              "details": "Engine failure at 33 seconds and loss of vehicle",
              "id": 2,
              "landSuccess": false,
              "name": "Falcon 2",
              "reddit": false,
              "reused": true,
              "type": "Merlin B",
            },
          ]
    }
    storeFilteredLaunchIds={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="refreshIcon"
          onClick={[Function]}
>
          <img
                    alt="refresh icon"
                    src="refresh.svg"
          />
</div>,
        <div
          className="checkboxes"
>
          <label
                    className="container landSuccess"
          >
                    LAND SUCCESS
                    <input
                              type="checkbox"
                    />
                    <span
                              className="checkmark"
                              onClick={[Function]}
                    />
          </label>
          <label
                    className="container reused"
          >
                    REUSED
                    <input
                              type="checkbox"
                    />
                    <span
                              className="checkmark"
                              onClick={[Function]}
                    />
          </label>
          <label
                    className="container withReddit"
          >
                    WITH REDDIT
                    <input
                              type="checkbox"
                    />
                    <span
                              className="checkmark"
                              onClick={[Function]}
                    />
          </label>
</div>,
      ],
      "className": "HeaderFilters",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="refresh icon"
            src="refresh.svg"
/>,
          "className": "refreshIcon",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "refresh icon",
            "src": "refresh.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              className="container landSuccess"
>
              LAND SUCCESS
              <input
                            type="checkbox"
              />
              <span
                            className="checkmark"
                            onClick={[Function]}
              />
</label>,
            <label
              className="container reused"
>
              REUSED
              <input
                            type="checkbox"
              />
              <span
                            className="checkmark"
                            onClick={[Function]}
              />
</label>,
            <label
              className="container withReddit"
>
              WITH REDDIT
              <input
                            type="checkbox"
              />
              <span
                            className="checkmark"
                            onClick={[Function]}
              />
</label>,
          ],
          "className": "checkboxes",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "LAND SUCCESS",
                <input
                  type="checkbox"
/>,
                <span
                  className="checkmark"
                  onClick={[Function]}
/>,
              ],
              "className": "container landSuccess",
            },
            "ref": null,
            "rendered": Array [
              "LAND SUCCESS",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "type": "checkbox",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "checkmark",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
            ],
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "REUSED",
                <input
                  type="checkbox"
/>,
                <span
                  className="checkmark"
                  onClick={[Function]}
/>,
              ],
              "className": "container reused",
            },
            "ref": null,
            "rendered": Array [
              "REUSED",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "type": "checkbox",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "checkmark",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
            ],
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "WITH REDDIT",
                <input
                  type="checkbox"
/>,
                <span
                  className="checkmark"
                  onClick={[Function]}
/>,
              ],
              "className": "container withReddit",
            },
            "ref": null,
            "rendered": Array [
              "WITH REDDIT",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "type": "checkbox",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "checkmark",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
            ],
            "type": "label",
          },
        ],
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="refreshIcon"
            onClick={[Function]}
>
            <img
                        alt="refresh icon"
                        src="refresh.svg"
            />
</div>,
          <div
            className="checkboxes"
>
            <label
                        className="container landSuccess"
            >
                        LAND SUCCESS
                        <input
                                    type="checkbox"
                        />
                        <span
                                    className="checkmark"
                                    onClick={[Function]}
                        />
            </label>
            <label
                        className="container reused"
            >
                        REUSED
                        <input
                                    type="checkbox"
                        />
                        <span
                                    className="checkmark"
                                    onClick={[Function]}
                        />
            </label>
            <label
                        className="container withReddit"
            >
                        WITH REDDIT
                        <input
                                    type="checkbox"
                        />
                        <span
                                    className="checkmark"
                                    onClick={[Function]}
                        />
            </label>
</div>,
        ],
        "className": "HeaderFilters",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="refresh icon"
              src="refresh.svg"
/>,
            "className": "refreshIcon",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "refresh icon",
              "src": "refresh.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                className="container landSuccess"
>
                LAND SUCCESS
                <input
                                type="checkbox"
                />
                <span
                                className="checkmark"
                                onClick={[Function]}
                />
</label>,
              <label
                className="container reused"
>
                REUSED
                <input
                                type="checkbox"
                />
                <span
                                className="checkmark"
                                onClick={[Function]}
                />
</label>,
              <label
                className="container withReddit"
>
                WITH REDDIT
                <input
                                type="checkbox"
                />
                <span
                                className="checkmark"
                                onClick={[Function]}
                />
</label>,
            ],
            "className": "checkboxes",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "LAND SUCCESS",
                  <input
                    type="checkbox"
/>,
                  <span
                    className="checkmark"
                    onClick={[Function]}
/>,
                ],
                "className": "container landSuccess",
              },
              "ref": null,
              "rendered": Array [
                "LAND SUCCESS",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "type": "checkbox",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "checkmark",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "span",
                },
              ],
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "REUSED",
                  <input
                    type="checkbox"
/>,
                  <span
                    className="checkmark"
                    onClick={[Function]}
/>,
                ],
                "className": "container reused",
              },
              "ref": null,
              "rendered": Array [
                "REUSED",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "type": "checkbox",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "checkmark",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "span",
                },
              ],
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "WITH REDDIT",
                  <input
                    type="checkbox"
/>,
                  <span
                    className="checkmark"
                    onClick={[Function]}
/>,
                ],
                "className": "container withReddit",
              },
              "ref": null,
              "rendered": Array [
                "WITH REDDIT",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "type": "checkbox",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "checkmark",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "span",
                },
              ],
              "type": "label",
            },
          ],
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
